### ISSUE TITLE
Gateway time-out when opening a long relation history

### ISSUE THREAD

**Issue by gy-mate** [openstreetmap/operations#1164]
### URL

https://www.openstreetmap.org/relation/7795942/history

### How to reproduce the issue?

1. Open the URL
2. See the following error:
    <img width="1440" alt="Screenshot 2024-10-23 at 23 47 36" src="https://github.com/user-attachments/assets/2f5c66a8-8966-4b5c-b214-37daa0a2ccc1">

### Screenshot(s) or anything else?

This relation is enormous: it has 275 members and 94 versions. But the website could handle these cases somehow (error handling / longer timeout / splitting versions to several pages), I think, considering that the relation of the Hungarian railway line _1_ has more members and might also reach this version size in the near future: https://www.openstreetmap.org/relation/11965670

**Comment by Firefishy** [comment]
Cloudflare timeout is 100 seconds and we are unable to change it in the edition we are using.

**Comment by gy-mate** [comment]
@Firefishy For me it times out in just _30_ seconds, not 100.

**Comment by AntonKhorev** [comment]
Maybe Cloudflare decides to display their timeout page when it receives a timeout status code from osm-website?

**Comment by gy-mate** [comment]
> Cloudflare timeout is 100 seconds and we are unable to change it in the edition we are using.

@Firefishy For me it still times out in just **30** seconds, not 100.

**Comment by tomhughes** [comment]
Which is the duration of the upstream timeout so I guess that is firing but cloudflare is replacing the rails error with it's own.

**Comment by gy-mate** [comment]
I see. Would dividing the versions loaded on the relation history page into pages of, for instance, 20, resolve this issue? (So that the PostgreSQL query would be sped up with `LIMIT` and `OFFSET`.)

https://www.openstreetmap.org/relation/11965670/history is also getting close to the timeout: currently it loads in ~25 seconds.

**Comment by gy-mate** [comment]
This was already suggested at https://github.com/openstreetmap/openstreetmap-website/issues/3524. I guess that would fix this issue.

If you don't mind, I'll close this one as a duplicate.

**Comment by gy-mate** [comment]
https://github.com/openstreetmap/openstreetmap-website/issues/3524 solved the issue for the relations below:

- https://www.openstreetmap.org/relation/7795942/history
- https://www.openstreetmap.org/relation/11965670/history
- https://www.openstreetmap.org/relation/544260/history
- https://www.openstreetmap.org/relation/197845/history
- https://www.openstreetmap.org/relation/2769137/history

But unfortunately https://www.openstreetmap.org/relation/9818928/history still shows a _504 Gateway Timeout_ Cloudflare error page after 30 seconds. :(

**Comment by tomhughes** [comment]
Yes if people created stupid relations you're probably not going to be able to view them on the web site because we have timeouts in place to protect the service.

**Comment by gy-mate** [comment]
Good point—I've deleted that relation (and two other related ones) because they are disused according to [FlixBus](https://shop.flixbus.hu/search?departureCity=40de1ad1-8646-11e6-9066-549f350fcb0c&arrivalCity=40e28c6a-8646-11e6-9066-549f350fcb0c&route=Prága-Szlavonszkibród&rideDate=17.10.2025&adult=1&_locale=hu&departureCountryCode=CZ&arrivalCountryCode=HR&features%5Bfeature.enable_distribusion%5D=1&features%5Bfeature.train_cities_only%5D=0&features%5Bfeature.auto_update_disabled%5D=0&features%5Bfeature.webc_search_us_veterans_promoted%5D=0&features%5Bfeature.webc_search_no_stations_limit%5D=0&features%5Bfeature.webc_station_search%5D=0&features%5Bfeature.webc_search_grouping_trips%5D=0&features%5Bfeature.darken_page%5D=1): https://www.openstreetmap.org/changeset/170764169
